src/lib.rs

```diff
@@ -26,12 +24,12 @@ pub fn open_sockets(source_peers: usize) -> Arc<Mutex<Vec<Option<TcpStream>>>> {
 
 /// Construct replayers that read data from sockets and can stream it into
 /// timely dataflow.
-pub fn make_replayers(sockets: Arc<Mutex<Vec<Option<TcpStream>>>>, index: usize, peers: usize) -> Vec<EventReader<Product<RootTimestamp, u64>, (u64, TimelySetup, TimelyEvent), TcpStream>> {
+pub fn make_replayers(sockets: Arc<Mutex<Vec<Option<TcpStream>>>>, index: usize, peers: usize) -> Vec<EventReader<u64, (u64, TimelySetup, TimelyEvent), TcpStream>> {
 
     sockets.lock().unwrap()
         .iter_mut().enumerate()
         .filter(|(i, _)| *i % peers == index)
         .map(move |(_, s)| s.take().unwrap())
-        .map(|r| EventReader::<Product<RootTimestamp, u64>, (u64, TimelySetup, TimelyEvent),_>::new(r))
+        .map(|r| EventReader::<u64, (u64, TimelySetup, TimelyEvent),_>::new(r))
```

Comment:
@comnik are you sure about this type?

Reply:
This *probably* wants to be a `Duration` rather than a `u64`, in both locations. What I have locally (not sure why not pushed, but in branch `logging`) is:

```rust
/// Construct replayers that read data from sockets and can stream it into
/// timely dataflow.
pub fn make_replayers(sockets: Arc<Mutex<Vec<Option<TcpStream>>>>, index: usize, peers: usize) -> Vec<EventReader<Duration, (Duration, WorkerIdentifier, TimelyEvent), TcpStream>> {

    sockets.lock().unwrap()
        .iter_mut().enumerate()
        .filter(|(i, _)| *i % peers == index)
        .map(move |(_, s)| s.take().unwrap())
        .map(|r| EventReader::<Duration, (Duration, WorkerIdentifier, TimelyEvent),_>::new(r))
        .collect::<Vec<_>>()
}
```

Reply:
That does the trick, just spotted that, didn't hink of verifying the actual type before...